### ISSUE TITLE
superset helm uses dockerize image which is unsupported (last release 7 years ago) and has CVEs

### ISSUE THREAD

**Issue by tooptoop4** [apache/superset#23103]
https://github.com/apache/superset/releases/download/superset-helm-chart-0.8.6/superset-0.8.6.tgz has references to jwilder/dockerize image/commands

https://github.com/jwilder/dockerize shows no new release for many years

following tools/CVEs are found within the dockerize image:

busybox 1.26.2-r4 has
CVE-2015-9261
CVE-2017-15873
CVE-2017-16544
CVE-2018-1000500
CVE-2018-1000517
CVE-2018-20679
CVE-2019-5747
CVE-2021-42376
CVE-2021-42378
CVE-2021-42379
CVE-2021-42381
CVE-2021-42382
CVE-2021-42384
CVE-2021-42385
CVE-2021-42386
CVE-2022-28391

musl 1.1.16-r9 has 
CVE-2017-15650
CVE-2019-14697
CVE-2020-28928

**Comment by dpgaspar** [comment]
> @tooptoop4 I see https://github.com/jwilder/dockerize has merged PRs into the repo more recently now and its most recent release seems to be in May 2023. Is this still a concern or can this issue be closed?

Still a concern, we should eventually remove this dependency, I think we just use it to test PG and redis availability on helm

**Comment by rusackas** [comment]
Does anyone here (cc @mistercrunch) know of any viable alternatives?

**Comment by mistercrunch** [comment]
While working on docker-related things I remember thinking "what does this do!?", and it's something about waiting for services to get up before starting other services. Asking GPT there are drop-in replacements like `wait-for-it` and `dockerize-golang` but given our experience with dockerize, I think the best would be to go helm or k8s-native with something like:

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: superset
spec:
  replicas: 1
  template:
    metadata:
      labels:
        app: superset
    spec:
      initContainers:
      - name: wait-for-db
        image: busybox
        command: ['sh', '-c', 'until nc -z db 5432; do echo waiting for db; sleep 2; done;']
      - name: wait-for-redis
        image: busybox
        command: ['sh', '-c', 'until nc -z redis 6379; do echo waiting for redis; sleep 2; done;']
      containers:
      - name: superset
        image: apache/superset:latest
        ports:
        - containerPort: 8088
```

**Comment by villebro** [comment]
@mistercrunch I agree, this would be much cleaner and easier to reason about. Another alternative would be to have a few new scripts (similar to `superset_init.sh`) for checking metastore/cache readiness, and then just use the superset image to avoid pulling in `busybox` (after all, it'll already be available). But I don't see `busybox` going sour or bloating anytime soon, so I think this is a really good approach.

**Comment by tooptoop4** [comment]
🦕

**Comment by mistercrunch** [comment]
Can't allocate time personally, but someone should tackle.

**Comment by rusackas** [comment]
Is this still the case with the current Helm chart?

**Comment by rusackas** [comment]
Can anyone confirm whether the current Helm chart has been updated to replace the old `dockerize` image? @rusackas asked this in July 2025. If it's been updated, this can be closed.

**Comment by bouhalimedaziz** [comment]
@rusackas I took a look at the current helm/superset/values.yaml on master, the initImage is using repository: apache/superset with tag: dockerize, so it's no longer pointing to the old jwilder/dockerize image. That said, dockerize itself is still there, just maintained by the Superset team now rather than fully replaced with a k8s-native approach like proposed in earlier messages.
https://github.com/apache/superset/blob/master/helm/superset/values.yaml

**Comment by rusackas** [comment]
@villebro @sha174n @dpgaspar might be best qualified to handle this.... if it's even still relevant. I'll tun Trivy and see what's up.

**Comment by rusackas** [comment]
Closing in favor of #40424, which captures the current state with a fresh trivy scan and a concrete proposal.

Quick summary of what changed since this was opened in Feb 2023:

- The chart no longer uses the upstream `jwilder/dockerize` image — it's been on `apache/superset:dockerize` (Superset-built) for a while. So the *specific* concern (3rd-party unmaintained image) is addressed.
- However, today's trivy scan of the current published `apache/superset:dockerize` shows we've taken on the maintenance burden and haven't kept up:

  | | |
  |---|---|
  | Image created | 2024-05-09 |
  | Base | Alpine 3.19.1 (EOSL) |
  | CRITICAL | 3 |
  | HIGH | 25 |
  | MEDIUM | 71 |
  | LOW | 24 |
  | Fixes available | 123/123 |

  64 of the CVEs are in the `dockerize` Go binary itself (stale Go stdlib + vendored `golang.org/x/{net,crypto}`); the rest are alpine base packages.

Rather than rebuild a single-purpose image on a fresher base (kicks the can again), #40424 proposes dropping `dockerize` from the chart entirely and replacing the four init-container TCP waits with a bash `/dev/tcp` probe against the superset image we're already pulling. Zero new images, no external deps, no CVE backlog separate from the main image's.

Thanks to everyone who weighed in here over the past three years — the proposal in #40424 is essentially @villebro's suggestion from June 2024 with a concrete migration plan.

🦕
